### PR TITLE
Rename ord_at to unicode_at for String

### LINKED ISSUE #15519: Rename String::ord_at to String::unicode_at
**Godot version:** Godot 3.0

The function name  `unicode_at` should more meaningful than `ord_at `.
I think it is the good chance to make this rename before the 3.0 release.

### PR DESCRIPTION
**Pull request by Geequlim**
Close #15519

### CODE REVIEW

File: doc/classes/String.xml [hunk 1]
```diff
@@ -530,13 +530,13 @@
...
-		<method name="ord_at">
+		<method name="unicode_at">
 			<return type="int">
 			</return>
-			<argument index="0" name="at" type="int">
+			<argument index="0" name="idx" type="int">
 			</argument>
 			<description>
-				Returns the character code at position [code]at[/code].
+			        Returns the character code at position [code]idx[/code].
```
Comment (akien-mga) [inline doc/classes/String.xml]: You changed the tab to spaces.

**Comment by akien-mga** [comment]
GDNative fails building it seems:
```
modules/gdnative/gdnative_api_struct.gen.cpp:706:2: error: 'godot_string_ord_at' was not declared in this scope
  godot_string_ord_at,
  ^
```

File: doc/classes/String.xml (hunk 1 above)
Comment (Geequlim) [reply doc/classes/String.xml]: sorry clang-formate doesn't work for me on Windows. I will fix it when I get home

**Comment by akien-mga** [comment]
I don't think it's worth breaking the API this late in the release cycle, and we only accept fixes to critical issues now (+ some enhancements to recent features like mono, bullet, gdnative, autotile, etc.).

We can consider it for 3.1 / 3.0.x, but then it will have to be added in a way that does not break compatibility (though `ord_at` should be kept, albeit with a deprecation warning announcing when it will be removed and pointing to the new method).

File: doc/classes/String.xml (hunk 1 above)
Comment (akien-mga) [reply doc/classes/String.xml]: `clang-format` doesn't apply to the XML, for those you need to preserve the formatting manually.

**Comment by Geequlim** [comment]
@akien-mga  Is there a way to flag a method is deprecated?

**Comment by akien-mga** [comment]
Not yet, but there will be when we implement #4397.

**Comment by akien-mga** [comment]
Closing for now as we need a proper deprecation workflow before this can be implemented. I've listed it in #16863, so we can rediscuss it and maybe make the change later on.

**Pull request closed without merging**